exir/passes/debug_handle_generator_pass.py

```diff
@@ -4,31 +4,75 @@
 # This source code is licensed under the BSD-style license found in the
 # LICENSE file in the root directory of this source tree.
 
+from typing import Dict
+
 from executorch.exir.graph_module import bfs_trace_with_node_process
 from executorch.exir.pass_base import ExportPass
 from torch.export import ExportedProgram
-from torch.fx import GraphModule
+from torch.fx import GraphModule, Node
 from torch.fx.passes.infra.pass_base import PassResult
 
 
 class DebugHandleGeneratorPass(ExportPass):
     def call(self, graph_module: GraphModule) -> PassResult:
-        """Lower a quantized reference model (with reference quantized operator patterns)
-        to executorch backend, that has a canonical set of quantized operators
+        """Generate debug handles for each node in the graph module and its submodule except
+        placeholder and output nodes. The debug handle is generated starting from 1 and
+        incrementally. The debug handle of a node is the same as the node sharing the same
+        greatest ancestor node in the export flow.
         """
 
-        index = 1
+        FROM_NODE_KEY = "from_node"
+        DEBUG_HANDLE_KEY = "debug_handle"
+
+        source_node_id_to_debug_handle: Dict[str, int] = {}
+
+        def _get_greatest_ancestor_node_identifier(node: Node) -> str:
+            """Get the identifier of the greatest ancestor node of the given node.
+
+            The identifier is the concatenation of the node name and graph id of the
+            greatest ancestor node, where the graph id is the unique id for every graph
+            module in the export flow and node name is unique within the same graph module.
+            """
+
+            node_source = node.meta[FROM_NODE_KEY]
+            node_source = node_source[-1]
```

Comment:
this is only populated for unlifted graph when `.module` is called? if so how is it guranteed that this is present

Reply:
no it is not only populate that `.module` is explicitly called. In our case it is guarantee generated after run_decompostition so we will have that when generating debug handle.

Reply:
> it is guarantee generated after run_decompostition so we will have that when generating debug handle.

There in my question around to_edge_transform_and_lower. although that specifically is unrelated to this pr